main/views_test.py

```diff
@@ -50,20 +59,24 @@ def test_js_settings(settings, client, is_authenticated):
     if is_authenticated:
         client.force_login(user)
 
-    response = client.get(reverse("main-index"))
-
-    js_settings = json.loads(response.context["js_settings_json"])
-    assert js_settings == {
-        "gaTrackingID": "fake",
-        "public_path": "/static/bundles/",
-        "environment": settings.ENVIRONMENT,
-        "sentry_dsn": "",
-        "release_version": settings.VERSION,
-        "user": {
-            "username": user.username,
-            "email": user.email,
-            "name": user.name,
+    response = client.get(reverse(name))
+    if expected_success:
+        assert response.status_code == HTTP_200_OK
+        js_settings = json.loads(response.context["js_settings_json"])
+        assert js_settings == {
+            "gaTrackingID": "fake",
+            "public_path": "/static/bundles/",
+            "environment": settings.ENVIRONMENT,
+            "sentry_dsn": "",
+            "release_version": settings.VERSION,
+            "user": {
+                "username": user.username,
+                "email": user.email,
+                "name": user.name,
+            }
+            if is_authenticated
+            else None,
         }
-        if is_authenticated
-        else None,
-    }
+    else:
+        assert response.status_code == HTTP_302_FOUND
+        assert response.url.startswith("/?next=")
```

Comment:
should we have an assert for `js_settings` here too, or is that not important?

Reply:
I don't think there would be anything there since it's a redirect. The url being redirected to already has `js_settings` asserted as part of the `main-index` case here, though we may want to update that if we create a specific login page that we redirect to